src/ChangeStream.php

```diff
@@ -214,6 +192,38 @@ private function isResumableError(RuntimeException $exception)
         return true;
     }
 
+    /**
+     * Perform housekeeping after an iteration event (i.e. next or rewind).
+     *
+     * @param boolean $isNext Whether the iteration event was a call to next()
+     * @throws ResumeTokenException
+     */
+    private function onIteration($isNext)
+    {
+        /* If the cursorId is 0, the server has invalidated the cursor and we
+         * will never perform another getMore nor need to resume since any
+         * remaining results (up to and including the invalidate event) will
+         * have been received in the last response. Therefore, we can unset the
+         * resumeCallable. This will free any reference to Watch as well as the
+         * only reference to any implicit session created therein. */
```

Comment:
This comment was also revised a bit to better explain why we do this. I shared some thoughts on this in Slack and @p-mongo chimed in with a sanity check.